From 001bcf044f49a717f7fdb5a8d84027e4dd5bc5ec Mon Sep 17 00:00:00 2001
From: Gabriel Arazas <foodogsquared@foodogsquared.one>
Date: Sat, 28 Feb 2026 12:12:18 +0800
Subject: [PATCH] Fix CMakeLists for mupdf library

For now, since all the author does with it is fetching the latest mupdf
library, we may as well have it from our own source.
---
 CMakeLists.txt | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b4a8f42..b2db349 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,6 +29,7 @@ set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g -fno-omit-frame-pointer -DNDEBUG")
 
 find_package(Qt6 REQUIRED COMPONENTS Gui Widgets Core Concurrent) #OpenGLWidgets )
 find_package(PkgConfig REQUIRED)
+pkg_check_modules(MUPDF REQUIRED mupdf)
 
 # SyncTeX
 find_path(SYNCTEX_INCLUDE_DIR
@@ -126,11 +127,6 @@ endif()
 
 qt_add_resources(RESOURCES resources.qrc)
 
-# Add MuPDF headers
-target_include_directories(${PROJECT_NAME} PRIVATE
-    ${CMAKE_SOURCE_DIR}/external/mupdf/include
-)
-
 target_sources(${PROJECT_NAME} PRIVATE ${RESOURCES})
 
 if(NOT SYNCTEX_INCLUDE_DIR OR NOT SYNCTEX_LIBRARY)
@@ -142,14 +138,11 @@ else()
     target_link_libraries(${PROJECT_NAME} PRIVATE synctex)
 endif()
 
-
-
 target_link_libraries(${PROJECT_NAME} PRIVATE
     Qt6::Gui Qt6::Widgets Qt6::Core Qt6::Concurrent
-    ${CMAKE_SOURCE_DIR}/external/mupdf/build/release/libmupdf.a
-    ${CMAKE_SOURCE_DIR}/external/mupdf/build/release/libmupdf-third.a
+    ${MUPDF_LIBRARIES}
 )
-
+target_include_directories(${PROJECT_NAME} PUBLIC ${MUPDF_INCLUDE_DIRS})
 
 install(TARGETS ${PROJECT_NAME} DESTINATION bin)
 #Install icons to correct freedesktop - compliant locations
-- 
2.52.0

